### PR TITLE
[keyboard]Graystudio Aero75 Hotswap support

### PR DESCRIPTION
**Pull request by edwardslau**
## Description

Aero75 Hotswap PCB (ANSI layout)
## Types of Changes

- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gray_studio/aero75/keymaps/vial/config.h [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#pragma once
```
Comment (keyboard-magpie) [inline keyboards/gray_studio/aero75/keymaps/vial/config.h:1]: Unfortunately PRs to QMK main will not be accepted with any vial elements, these must be removed

Comment (edwardslau) [reply keyboards/gray_studio/aero75/keymaps/vial/config.h]: @keyboard-magpie Thank you for informing me, I have removed the vial part

File: keyboards/gray_studio/aero75/aero75.c [hunk 2]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (drashna) [inline keyboards/gray_studio/aero75/aero75.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/gray_studio/aero75/aero75.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (drashna) [inline keyboards/gray_studio/aero75/aero75.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/gray_studio/aero75/config.h [hunk 4]
```diff
@@ -0,0 +1,196 @@
+
```
Comment (drashna) [inline keyboards/gray_studio/aero75/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/gray_studio/aero75/config.h [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4753
+#define PRODUCT_ID      0x3075
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Graystudio
+#define PRODUCT         Aero 75
```
Comment (drashna) [inline keyboards/gray_studio/aero75/config.h:12]: Please move to info.json.

File: keyboards/gray_studio/aero75/config.h [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+#define RGBLIGHT_LAYERS_OVERRIDE_RGB_OFF
+#ifdef RGB_DI_PIN 
+#define RGBLED_NUM 1
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#define RGBLIGHT_LIMIT_VAL 200 /* The maximum brightness level */
+#define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+// /*== all animations enable ==*/
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/gray_studio/aero75/config.h:48]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/gray_studio/aero75/config.h [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
+/* Bootmagic Lite key configuration */
+// #define BOOTMAGIC_LITE_ROW 0
+// #define BOOTMAGIC_LITE_COLUMN 0
```
Comment (drashna) [inline keyboards/gray_studio/aero75/config.h:196]:
```suggestion
```

File: keyboards/gray_studio/aero75/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,55 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/gray_studio/aero75/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/gray_studio/aero75/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,55 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/gray_studio/aero75/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/gray_studio/aero75/rules.mk [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/gray_studio/aero75/rules.mk:21]: Please remove the `SLEEP_LED_ENABLE` setting, as it's not really used, and remove the comments for both it and NKRO.  
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/gray_studio/aero75/rules.mk [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (drashna) [inline keyboards/gray_studio/aero75/rules.mk:28]:
```suggestion
AUDIO_ENABLE = no           # Audio output
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/gray_studio/aero75/config.h (hunk 5 above)
Comment (edwardslau) [reply keyboards/gray_studio/aero75/config.h]: Can you specify this one?

Comment (edwardslau) [reply keyboards/gray_studio/aero75/config.h]: > Please move to info.json.

Can you specify this one? @drashna

Comment (keyboard-magpie) [reply keyboards/gray_studio/aero75/config.h:13]: As documented in the [PR guide](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs) these defines need to live in info.json

Comment (edwardslau) [reply keyboards/gray_studio/aero75/config.h]: > As documented in the [PR guide](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs) these defines need to live in info.json

Thank you for clarifying! I have made the changes. Can you please take a look? @keyboard-magpie

**Review by keyboard-magpie — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/gray_studio/aero75/rules.mk [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
```
Comment (drashna) [inline keyboards/gray_studio/aero75/rules.mk:21]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes      # Mouse keys
EXTRAKEY_ENABLE = yes      # Audio control and System control
CONSOLE_ENABLE = no        # Console for debug
COMMAND_ENABLE = no        # Commands for debug and configuration
NKRO_ENABLE = no           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes      # Enable keyboard RGB underglow
AUDIO_ENABLE = no          # Audio output
```

**Pull request merged**